### PR TITLE
Add support to HMR

### PR DESCRIPTION
**Pull request by FabioAntunes**
This is a fix for #10 

The reason why that error happens in the first place is because eveyrtime there's an HMR, the tree is recalculated, and since `currentId` and the `components`, don't get a reset on the server side, it keeps adding to those variables.

When a page is refreshed on the fronted, the tracker starts empty, so `currentId = 0` and `components= {}` creating a mismatch on the ids of the components from the backend to the frontend.

small example, if we have this routes.js:

```
// Routes.js
import loadable from 'loadable-components'

export const Home = loadable(() => import('./Home'))
export const About = loadable(() => import('./About'))
export const Contact = loadable(() => import('./Contact'))
```

**componentTracker.js** will be:
```
currentId = 3
components = {
  0: loadableFunc, // Home
  1: loadableFunc, // About
  2: loadableFunc, // Contact
}
```

if there's a HMR event on the server and frontend it will be: 

```
currentId = 6
components = {
  0: loadableFunc, // Home
  1: loadableFunc, // About
  2: loadableFunc, // Contact
  3: loadableFunc, // Home
  4: loadableFunc, // About
  5: loadableFunc, // Contact 
}
```

As long as we don't refresh the front end they will both be in sync, as soon we do a refresh, for example on the route  `/home`

server's  *componentTracker.js*:

```
currentId = 6
components = {
  0: loadableFunc, // Home
  1: loadableFunc, // About
  2: loadableFunc, // Contact
  3: loadableFunc, // Home
  4: loadableFunc, // About
  5: loadableFunc, // Contact 
}
```

it will return the id 3, because it's our current id for the **HomeComponent**:

But on the frontend our **componentTracker** will be:
```
components = {
  0: loadableFunc, // Home
  1: loadableFunc, // About
  2: loadableFunc, // Contact
}
```

And that's why we get an exception.

Now the function itself isn't the solution but it's part of it, the next step is simple adding this to our HMR part on the server and on the client:

**myServer.js**
```
import { loadableHMR } from 'loadable-components/componentTracker';
import app from './server';

if (module.hot) {
  module.hot.accept('./server', () => {
    console.log('🔁  HMR Reloading `./server`...');
    loadableHMR();
  });

  console.info('✅  Server-side HMR Enabled!');
}
```

**myClient.js**
```
import { loadableHMR } from 'loadable-components/componentTracker';

if (module.hot) {
  module.hot.accept();
  // hot.accept doesn't accept a callback, when the first arg is undefined  so we use the dispose
  module.hot.dispose(() => {
    loadableHMR();
  });
}
```

Basically what's happening in there, is whenever there's a HMR, just reset our componentTracker, so everyone will be in sync again. It may also prevent memory leaks in the future if we had zillions of HMR events along with zillions of loadable components

PS:
Do you want me to add a simple test for the function as this might decrease code coverage?

PPS:
Added tests anyway

### CODE REVIEW

**Comment by neoziro** [comment]
Thanks for this PR! I would like to wait for React Hot Loader v4 to be released before merging this approach. I would like a zero-config approach to make it works out of the box with React Hot Loader v4.

**Comment by FabioAntunes** [comment]
That's understandable since this is just a workaround, but at least it makes development easier until Hot loader v4 get's released. Right now it 's quite annoying to kill the server every time. 

This PR just allows to reset the counter manually, it doesn't adds any breaking changes or messes with the API internals.

Any predictions when hot loader v4 will be released?

**Comment by neoziro** [comment]
In one or two weeks I hope!

**Comment by neoziro** [comment]
I don't want to add code in this library to support HMR, if you want it, please use the next version of [React Hot Loader](https://github.com/gaearon/react-hot-loader/tree/next).
